### PR TITLE
Ctw 748/allergies update

### PR DESCRIPTION
**Pull request by pmadria3**
CTW-748

This PR adds updates to allergies table, sorting/filtering, and updating the details drawer. 

Screenshot 2023-04-03 at 1 51 42 PM

### CODE REVIEW

File: src/components/content/allergies/allergies-filter.tsx [hunk 1]
```diff
@@ -11,10 +17,8 @@ export const applyAllergyFilters = (data: fhir4.AllergyIntolerance[]) => {
...
 const valuesToDedupeOn = (allergy: AllergyModel) => [
-  allergy.categories,
-  allergy.clinicalStatus,
   allergy.display,
-  allergy.manifestations,
-  allergy.onset,
-  allergy.type,
+  allergy.recordedDate,
+  allergy.managingOrganization,
+  allergy.knownCodings,
```
Comment (eric-gunther) [inline src/components/content/allergies/allergies-filter.tsx:23]: The ticket AC says to de-dupe by code, preferring RxNorm over NDC over SNOMED, using the display value. So I think simply de-duping on `allergy.display` was correct, but we'll want to adjust that getter (will provide that feedback in a comment on that area of the code).

File: src/App.tsx [hunk 2]
```diff
@@ -105,7 +105,12 @@ const demoComponents: DemoComponent[] = [
     render: () => <PatientDocuments />,
     title: "Patient Documents",
   },
-  { render: () => <PatientAllergies />, title: "Patient Allergies" },
+  {
+    render: () => (
+      <ZusAggregatedProfile resources={["allergies"]} title="Allergies" />
+    ),
+    title: "Patient Allergies",
+  },
```
Comment (eric-gunther) [inline src/App.tsx:113]: Non-blocking feedback, but this whole App.tsx is starting to get a bit messy again. Perhaps we could just use a single instance of `ZusAggregatedProfile` where folks can play around with the `resources` prop.

File: src/components/content/allergies/helpers/history.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  );
+
+  const searchParams: SearchParams = {
+    _include: ["AllergyIntolerance:patient"],
+  };
+
+  // If we have any known codings, then do an OR search.
+  // Otherwise fall back to searching for this single condition.
+  // That way, conditions that don't have any good codes to match on
+  // will only show themselves in the history.
```
Comment (eric-gunther) [inline src/components/content/allergies/helpers/history.tsx:32]: Copy/paste comment still referencing conditions.

File: src/components/content/allergies/helpers/history.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  if (tokens.length > 0) {
+    searchParams.code = tokens.join(",");
+  } else {
+    // eslint-disable-next-line no-underscore-dangle
+    searchParams._id = allergy.id;
+  }
```
Comment (eric-gunther) [inline src/components/content/allergies/helpers/history.tsx:38]: You should investigate the performance of these queries by code in production, I suspect they are going to be slow (which I think I noticed for conditions already in prod).

File: src/components/content/allergies/helpers/history.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    date: allergy.recordedDate,
+    versionId: allergy.versionId,
+    title: allergy.managingOrganization,
+    details: detailData,
+  };
+}
+
+const valuesToDedupeOn = (allergy: AllergyModel) => [
+  allergy.knownCodings.map((coding) => [coding.system, coding.code]),
+];
```
Comment (eric-gunther) [inline src/components/content/allergies/helpers/history.tsx:65]: These are the values to de-dupe on in the history, right? So I would expect this to include the date field and perhaps others.

**Review by eric-gunther — CHANGES_REQUESTED** [review]

File: src/components/content/allergies/patient-allergies.tsx [hunk 6]
```diff
@@ -1,12 +1,19 @@
+/* eslint-disable import/order */
```
Comment (pmadria3) [inline src/components/content/allergies/patient-allergies.tsx:1]: eslint messed up in editor so had to add this - please can someone remove this

File: src/fhir/models/allergies.ts [hunk 7]
```diff
@@ -1,6 +1,11 @@
+/* eslint-disable import/order */
```
Comment (pmadria3) [inline src/fhir/models/allergies.ts:1]: same as prev comment

File: src/App.tsx [hunk 8]
```diff
@@ -63,52 +50,12 @@ const demoComponents: DemoComponent[] = [
...
-    title: "Patient Conditions",
-  },
-  {
-    render: () => <PatientDocuments />,
-    title: "Patient Documents",
-  },
-  { render: () => <PatientAllergies />, title: "Patient Allergies" },
-  { render: () => <PatientImmunizations />, title: "Patient Immunizations" },
-  { render: () => <PatientSearch />, title: "Patient Search" },
-  { render: () => <PatientTimelineV2 />, title: "Patient Timeline" },
```
[the remark this answers is not in the document]
Comment (pmadria3) [reply src/App.tsx:111]: Eric had a comment on it above - we can just do one instance and then pass the resource and use that

**Review by luthfib — APPROVED** [review]

**Pull request merged**